### ISSUE TITLE
NullRef on CacheInit for random joiner

### ISSUE THREAD

**Issue by FluffierThanThou** [FluffierThanThou/WorkTab#40]
```
Exception in RimWorld.ThinkNode_ConditionalColonist TryIssueJobPackage: System.NullReferenceException: Object reference not set to an instance of an object
  at Fluffy_Tabs.PawnPrioritiesTracker.InitPriorityCache () [0x00000] in <filename unknown>:0 
  at Fluffy_Tabs.PawnPrioritiesTracker..ctor (Verse.Pawn pawn) [0x00000] in <filename unknown>:0 
  at Fluffy_Tabs.WorldObject_Priorities.WorkgiverTracker (Verse.Pawn pawn) [0x00000] in <filename unknown>:0 
  at Fluffy_Tabs.Widgets.Priorities (Verse.Pawn pawn) [0x00000] in <filename unknown>:0 
  at Fluffy_Tabs.Detours_WorkSettings+<>c__DisplayClass9_0.<_CacheWorkGiversInOrder>b__1 (RimWorld.WorkGiver wg) [0x00000] in <filename unknown>:0 
  at System.Linq.Enumerable+<CreateWhereIterator>c__Iterator1D`1[RimWorld.WorkGiver].MoveNext () [0x00000] in <filename unknown>:0 
  at System.Linq.Enumerable.Any[WorkGiver] (IEnumerable`1 source) [0x00000] in <filename unknown>:0 
  at Fluffy_Tabs.Detours_WorkSettings._CacheWorkGiversInOrder () [0x00000] in <filename unknown>:0 
  at RimWorld.Pawn_WorkSettings.get_WorkGiversInOrderEmergency () [0x00000] in <filename unknown>:0 
  at RimWorld.JobGiver_Work.TryGiveJob (Verse.Pawn pawn) [0x00000] in <filename unknown>:0 
  at Verse.AI.ThinkNode_JobGiver.TryIssueJobPackage (Verse.Pawn pawn) [0x00000] in <filename unknown>:0 
  at Verse.AI.ThinkNode_Priority.TryIssueJobPackage (Verse.Pawn pawn) [0x00000] in <filename unknown>:0 
Verse.Log:Error(String)
Verse.AI.ThinkNode_Priority:TryIssueJobPackage(Pawn)
RimWorld.ThinkNode_Conditional:TryIssueJobPackage(Pawn)
Verse.AI.ThinkNode_Priority:TryIssueJobPackage(Pawn)
Verse.AI.Pawn_JobTracker:DetermineNextJob(ThinkTreeDef&)
Verse.AI.Pawn_JobTracker:TryFindAndStartJob()
Verse.AI.Pawn_JobTracker:JobTrackerTick()
Verse.Pawn:Tick()
Verse.TickList:Tick()
Verse.TickManager:DoSingleTick()
Verse.TickManager:TickManagerUpdate()
Verse.Game:Update()
Verse.Root_Play:Update()
```

**Comment by FluffierThanThou** [comment]
I can't reproduce in a no-mods scenario. Looking for output_logs for situations where this happened.

**Comment by FluffierThanThou** [comment]
Probably fixed with v0.16.2.2
